src/utils/getDonationUrl.ts

```diff
@@ -1,17 +1,31 @@
 // calling this function before window is loaded may cause an error
-export const getDonationUrl = (id: string, token: string | null): string => {
+export const getDonationUrl = (
+  id: string,
+  token: string | null,
+  embed?: string | undefined,
+  callbackUrl?: string | undefined
+): string => {
   const country = localStorage.getItem('countryCode');
   const language = localStorage.getItem('language');
   let directGift = localStorage.getItem('directGift');
   if (directGift) {
     directGift = JSON.parse(directGift);
   }
-  const sourceUrl = `${
-    process.env.NEXT_PUBLIC_DONATION_URL
-  }/?to=${id}&callback_url=${
-    window.location.href
-  }&country=${country}&locale=${language}${
-    token ? '&token=' + token : ''
-  }&tenant=${process.env.TENANTID}${directGift ? '&s=' + directGift.id : ''}`;
+  const sourceUrl =
+    embed === 'true'
+      ? `${process.env.NEXT_PUBLIC_DONATION_URL}/?to=${id}&embed=${embed}&${
+          callbackUrl != undefined ? `callback_url=${callbackUrl}` : ''
+        }&country=${country}&locale=${language}${
+          token ? '&token=' + token : ''
+        }&tenant=${process.env.TENANTID}${
+          directGift ? '&s=' + directGift.id : ''
+        }`
+      : `${process.env.NEXT_PUBLIC_DONATION_URL}/?to=${id}&callback_url=${
+          window.location.href
+        }&country=${country}&locale=${language}${
+          token ? '&token=' + token : ''
+        }&tenant=${process.env.TENANTID}${
+          directGift ? '&s=' + directGift.id : ''
+        }`;
```

Comment:
Difficult to read. It might be better to separate the logic around `embed=true` outside the building of the `sourceUrl`.

Something like

```
  const callbackUrl = (embed === "true") ? callbackUrl : window.location.href;
```

and

```
  const sourceUrl = ....{callbackUrl !== undefined && `callback_url=${callbackUrl}`}
```